### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by ezraermy**
This pull request creates a database using PostgreSQL
The implementation includes the following changes:
- Installed PostgreSQL
- Created a vet_clinic database
- Added a table called animals.
- Query and update the animals table.

## Other essential requirements:
- Use Gitflow.
- Documented in a professional way.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -4,5 +4,6 @@ CREATE TABLE animals(
   date_of_birth DATE NOT NULL,
   espcape_attempts INT NOT NULL,
   neutered BOOLEAN NOT NULL,
-  weight_kg DECIMAL (5, 2) NOT NULL
+  weight_kg DECIMAL (5, 2) NOT NULL,
+  species VARCHAR (255) NOT NULL
```
Comment (victorokoroji) [inline schema.sql]: - [ ] I will strongly recommend you dont modify the table directly, but rather alter the table by adding the species column by running the command:

`ALTER TABLE animals ADD species varchar(255);`

File: queries.sql [hunk 2]
```diff
@@ -6,3 +6,32 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- SQL script that implement transaction
```
Comment (victorokoroji) [inline queries.sql]: - [ ] I can't seem to find any of the below queries in your submission. Kindly follow the project requirement step by step and add each step of the query in your queries.sql file

image

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @ezraermy, 

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Table created properly ✅ 


## Required Changes ♻️

- [ ] KIndly add screenshots of all your queries in the pull request summary.

_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (ezraermy) [reply queries.sql]: Hi @vickymarz Thank you very much for your helpful and insightful feedback which surely makes my project better.

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi @ezraermy 🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [ ] As previously stated, please include screenshots of all your queries in the pull request summary :+1:. It is important to include these screenshots as they provide a visual representation of the work you have done and make it easier for others to review and understand your changes.

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ezraermy** [comment]
@IngDavidTM  Thank you for your kind feedback. However, I am a bit confused about which screenshot I am unable to provide. According to the instruction, it requires to do screenshots on four occasions. I provided five screenshots since the last one needs two. transaction_update_species.png(1), transaction_delete_rollback.png(2), transaction_create_savepoint(3) and write_queries1&2(4).

Can you please be more specific about which screenshot in the given task is not included? If there is anything I missed to show on my screenshots? 

Thank you for your time once again.

**Comment by Gambit142** [comment]
Hi @ezraermy, kindly add screenshots for the following transactions that you wrote:

```SQL
BEGIN TRANSACTION;
UPDATE animals SET species = 'unspecified';
ROLLBACK;

BEGIN TRANSACTION;
UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
COMMIT;
```

#### Hint: After committing or rolling back the transactions, you can run this command:

`SELECT * from animals;`, then add the screenshot to your queries_image directory. 👍 😎

File: queries.sql [hunk 3]
```diff
@@ -6,3 +6,43 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+-- Classify species into digimon and pokemon 
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
```
Comment (Gambit142) [inline queries.sql]: - [ ] Based on the former reviewer's comment, you ought to add screenshots for the following transactions that you wrote:

```SQL
BEGIN TRANSACTION;
UPDATE animals SET species = 'unspecified';
ROLLBACK;

BEGIN TRANSACTION;
UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
COMMIT;
```

#### Hint: After committing or rolling back the transactions, you can run this command:

`SELECT * from animals;`, then add the screenshot to your queries_image directory. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @ezraermy 👋, 

Great work on making the changes requested by a previous reviewer 👏 

## Highlights
- Everything that creates or changes the structure of the table is in the schema.sql file ✔️
- Everything that populates tables with data is in the data.sql file ✔️
- Everything that is a query that helps to analyze is in the queries.sql file ✔️

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (ezraermy) [reply queries.sql]: @Gambit142 Thanks a lot for the feedback and tips. It is much clearer now.

**Review by andersonlebon — APPROVED** [review]
## Hi @ezraermy 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (Gambit142) [reply queries.sql]: @ezraermy, you are welcome. 👍

**Pull request merged**